### PR TITLE
[Enhancement] Optimize the log info for slow lock

### PR DESCRIPTION
**Pull request by nshangyiming**
## What type of PR is this:
- [ ] BugFix
- [ ] Feature
- [x] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Checklist:
- [x] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [x] 3.0
  - [x] 2.5
  - [ ] 2.4
  - [ ] 2.3

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 4 / 7 (57.14%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/catalog/Database.java|0|2|00.00%|[171, 172]|
|:large_blue_circle:|com/starrocks/clone/TabletScheduler.java|0|1|00.00%|[1776]|
|:large_blue_circle:|com/starrocks/common/util/LogUtil.java|4|4|100.00%|[]|

File: fe/fe-core/src/main/java/com/starrocks/common/util/LogUtil.java [hunk 1]
```diff
@@ -45,4 +49,11 @@ public static void logConnectionInfoToAuditLogAndQueryQueue(ConnectContext ctx,
         queryDetail.setErrorMessage(ctx.getState().getErrorMessage());
         QueryDetailQueue.addAndRemoveTimeoutQueryDetail(queryDetail);
     }
+
+    public static String getCurrentStackTrace() {
+        TimeoutException e = new TimeoutException("lock acquisition timeout");
```
Comment (kevincai) [inline fe/fe-core/src/main/java/com/starrocks/common/util/LogUtil.java:54]: should not use TimeoutException, if this is a generic method to grape the current thread trace, 

use `Thread.currentThread().getStackTrace();` or `new Throwable().getStackTrace();`

Comment (nshangyiming) [reply fe/fe-core/src/main/java/com/starrocks/common/util/LogUtil.java]: good point

**Review by kevincai — DISMISSED** [review]

**Review by kevincai — DISMISSED** [review]

**Review by gengjun-git — DISMISSED** [review]

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 3 / 6 (50.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/catalog/Database.java|0|2|00.00%|[171, 172]|
|:large_blue_circle:|com/starrocks/clone/TabletScheduler.java|0|1|00.00%|[1776]|
|:large_blue_circle:|com/starrocks/common/util/LogUtil.java|3|3|100.00%|[]|

**Review by gengjun-git — APPROVED** [review]

**Review by Astralidea — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-2.5

**Pull request merged**